drivers/watchdog/wdt_nrfx.c

```diff
@@ -124,6 +86,23 @@ static int wdt_nrf_install_timeout(struct device *dev,
 	}
 
 	if (m_allocated_channels == 0) {
+		/* install_timeout function passes millisecond values as a
+		 * window value in cfg structure.
+		 * Value passed to Nordic's peripheral has to be greater than
+		 * 15 ticks (see Product Specification for more) clocked by
+		 * 32768kHz frequency. It means that Nordic's watchdog requires
+		 * at least 0.458ms reload time.
+		 * Watchdog API allows to use only milliseconds. Code should
+		 * check if time is greater or equal 1ms (Nordic's watchdog
+		 * restriction rounded up to integer value).
+		 * TO sum up, in case of having zero as max window value
+		 * the function returns -EINVAL. Values greater or equal 1ms
+		 * are correct.
```

Comment:
Not entirely. Values > 0x07CFFFFF are not valid because the number of ticks for them will be > 0xFFFFFFFF.

Reply:
And I would shorten the comment to something like: "According to relevant Product Specifications, watchdogs in all nRF chips can use reload values (determining the timeout) from range 0xF-0xFFFFFFFF given in 32768 Hz clock ticks. Check if the provided value expressed in milliseconds results in a valid amount of ticks."